### PR TITLE
fix: use PEP440 for Python package version comparison

### PR DESCRIPTION
**Pull request by willmurphyscode**
There have been several issues opened where `certifi` Python package, which uses a version string like `2022.07.22`, which is PEP440 compliant but may or may not look like semver depending on the regex in use, results in incorrect version comparisons. Switch python version comparison to use Python PEP440 directly.

Fixes https://github.com/anchore/grype/issues/1430
Fixes https://github.com/anchore/grype/issues/1417
Fixes https://github.com/anchore/grype/issues/1172
Fixes https://github.com/anchore/grype/issues/1034
Fixes https://github.com/anchore/grype/issues/986

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,7 @@
 /.grype.yaml
 
 CHANGELOG.md
-VERSION
+/VERSION
```
Comment (willmurphyscode) [inline .gitignore:7]: One case insensitive file systems (such as macOS), this line previously resulted in ignoring files under `./grype/version/`.

File: grype/version/pep440_constraint_test.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+		{
+			name:       "date based pep440 version string boundary condition",
+			version:    "2022.12.7",
+			constraint: ">=2017.11.05,<2022.12.07",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := newPep440Constraint(tc.constraint)
+			require.NoError(t, err)
```
Comment (willmurphyscode) [inline grype/version/pep440_constraint_test.go:302]: Big `TODO` here - didn't actually implement these tests yet. Test cases are copied from the old fuzzy matcher.

**Review by kzantow — APPROVED** [review]
LGTM

**Review by wagoodman — APPROVED** [review]

**Comment by alex** [comment]
As a certifi maintainer: thank you!

File: grype/version/constraint.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ func GetConstraint(constStr string, format Format) (Constraint, error) {
 		// Although this will work in most cases, some oddities aren't supported, like:
 		// 1.0b2.post345.dev456 which is allowed by the spec. In that case (a dev release of a post release)
 		// the comparator will fail. See https://www.python.org/dev/peps/pep-0440
```
Comment (danielsamuels) [inline grype/version/constraint.go:27]: Presumably this comment is outdated now?

Comment (willmurphyscode) [reply grype/version/constraint.go:27]: Good catch @danielsamuels! I'll send a PR to clean up the comment. Thanks!
